src/main/java/org/dcsa/ebl/model/transferobjects/ShippingInstructionTO.java

```diff
@@ -2,122 +2,135 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
-import org.dcsa.core.events.model.Reference;
-import org.dcsa.core.events.model.transferobjects.LocationTO;
+import org.dcsa.core.events.model.transferobjects.*;
 import org.dcsa.ebl.model.base.AbstractShippingInstruction;
 
 import javax.validation.Valid;
-import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
 import java.util.List;
 
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
 public class ShippingInstructionTO extends AbstractShippingInstruction {
 
-    @NotNull
-    @Valid
-    private List<CargoItemTO> cargoItems;
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private String carrierBookingReference;
 
-    @NotNull
-    @Valid
-    private List<ShipmentEquipmentTO> shipmentEquipments;
+  private LocationTO placeOfIssue;
```

Comment:
Do we have a annotation-validation that we are using for the Locations?

Reply:
As far as I know we only have location validator for subtypes (unlocation, geolocation, facility, address).